Makefile

```diff
@@ -145,13 +162,15 @@ build_image: envvar
 	@echo "${GREEN}Makefile: Building image: ${TAG}${RESET}"
 ifeq ($(TAG),)
 	@echo "This is a sourced target!"
-	@echo "Do not run this target directly... exitting!"
+	@echo "Do not run this target directly... exiting!"
 	@exit 1
 endif
 	${DEBUG}if [ ! -e "./Dockerfile" ]; then \
 		IMAGE="`echo $${TAG} | sed -e s#\'##g -e s#localhost\/## -e s#:latest##`"; \
-		if [ "`curl https://raw.githubusercontent.com/kubevirt/project-infra/master/images/$${IMAGE}/Dockerfile -o Dockerfile -w '%{http_code}\n' -s`" != "200" ]; then \
-			echo "curl Dockerfile failed... exitting!"; \
+		DOCKERFILE_URL="https://raw.githubusercontent.com/kubevirt/project-infra/master/images/$${IMAGE}/Dockerfile" \
+		DOCKERFILE_GET_RESULT="`curl $${DOCKERFILE_URL} -o Dockerfile -w '%{http_code}\n' -s`"; \
+		if [ "$${DOCKERFILE_GET_RESULT}" != "200" ]; then \
+			echo "Download of Dockerfile from [$${DOCKERFILE_URL}] failed... exiting!"; \
```

Comment:
#170-172 ... I thought about doing this but setting two extra vars seemed meh.  Convince me.  :)

I really prefer not doing inline bash scripts but I couldn't get a Makefile rule to set env vars from command output.  I have a feeling the problem came from old GNU tools that Fedora/RHEL packs.

#173 is +1

Reply:
Yeah 170-172 got broken out primarily to make L173 easier to accomplish -- these make variables should only actually exist during the target execution, so they won't leak out anywhere else if I'm understanding `make` correctly. So I don't have better reasoning than that, wanted to be able to be specific about the failure but didn't want to copy that long URL.

We could make them actual top level variables (and thus modifiable by ENV) as well, since it looks like `?= $(shell ...)` would keep the right side "deferred" and ensure that `curl` only ran when it was needed -- I didn't do this initially because I was worried that simply running `make` for another target would run `curl` immediately.

Reply:
Ok, I'm convinced.  We can always revert later if we don't like the behavior.

I'll resolve this conversation.